### PR TITLE
Partial revert of battle calc simplifications

### PR DESCRIPTION
**Pull request by RoiEXLab**
This is a partial revert of #6328 which should fix the performance regression reported by @asvitkine and the issue reported in #6349

@asvitkine It would be nice if you could verify this actually solves the regression for you. It should, but there's also a slight chance it doesn't.

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[x] Problem fix:  #6349 https://github.com/triplea-game/triplea/pull/6328#issuecomment-620933476
[] Other:

### CODE REVIEW

File: game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java [hunk 1]
```diff
@@ -95,21 +234,34 @@ public AggregateResults calculate(
...
-                  individualRemaining ->
-                      createBattleCalcWorker(
+    synchronized (mutexCalcIsRunning) {
+      awaitLatch();
+      final long start = System.currentTimeMillis();
+      // Create worker thread pool and start all workers
+      int totalRunCount = 0;
+      int runCount = initialRunCount;
+      final int workerNum = workers.size();
+      final int workerRunCount = Math.max(1, (runCount / Math.max(1, workerNum)));
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java:246]: This code here changed slightly from the original implementation, basically we compute all of the run counts here now.
Basically it does the same as my IntStream implementation to the left with some added exit conditions

**Comment by asvitkine** [comment]
I patched this in at current tip of tree and compared with eb21301 which was before the regression and tried my saved game.

I'm seeing an even faster performance now than before - under 30s whereas closer to 45s before. (And with the regression it was several minutes I think.)

So this seems great!

**Comment by RoiEXLab** [comment]
@asvitkine That's cool to hear, the restructured interface allowed me to remove some unused code and intermediate steps (such as double synchronization on the same object), that could be one of the reasons for the increased performance if the code is called very often

**Comment by asvitkine** [comment]
Actually, maybe it's too good to be true? I tried the Odds calculator in game and it gave 0% odds for something that shouldn't be 0. So something is broken?

**Comment by RoiEXLab** [comment]
@asvitkine You're right, I accidentally dropped one line where a boolean flag was set. It doesn't make sense anymore so I can just remove it. Please check the latest commit

**Comment by asvitkine** [comment]
Tested the new patch. 45s - so similar to before the regression, just a bit faster.

**Pull request merged**